components/feature/prompts/src/main/java/mozilla/components/feature/prompts/logins/LoginValidationDelegate.kt

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.logins
+
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.async
+import mozilla.appservices.logins.InvalidLoginReason
+import mozilla.appservices.logins.InvalidRecordException
+import mozilla.appservices.logins.ServerPassword
+import mozilla.components.concept.engine.autofill.Login
+import mozilla.components.feature.prompts.logins.LoginValidationDelegate.Result
+import mozilla.components.lib.dataprotect.SecureAbove22Preferences
+import mozilla.components.service.sync.logins.AsyncLoginsStorage
+
+/**
+ * Provides a method for checking whether or not a given login can be stored.
+ */
+interface LoginValidationDelegate {
```

Comment:
Move this to `concept-storage`? (or `concept-login-autofill` if you really have to).